zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ZiplineServeTask.kt

```diff
@@ -50,18 +56,44 @@ abstract class ZiplineServeTask : DefaultTask() {
     val deploymentRegistry = services.get(DeploymentRegistry::class.java)
     val deploymentHandle = deploymentRegistry.get(deploymentId, ZiplineServerDeploymentHandle::class.java)
     if (deploymentHandle == null) {
-      val server = routes(
-        "/" bind static(Directory(inputDir.get().asFile.absolutePath), Pair("zipline", ContentType.TEXT_PLAIN))
-      ).asServer(SunHttp(port.orElse(8080).get()))
-
+      // First time this task is run, start the server
+      val ws = websockets(
+        "/ws" bind { ws: Websocket ->
+          websockets.add(ws)
+          ws.send(WsMessage(RELOAD_MESSAGE))
+          ws.onClose {
+            websockets.remove(ws)
+          }
+        }
+      )
+      val http = routes(
+        "/" bind static(
+          Directory(inputDir.get().asFile.absolutePath),
+          Pair("zipline", ContentType.TEXT_PLAIN)
+        )
+      )
+      val server = PolyHandler(http, ws).asServer(Jetty(8080))
       deploymentRegistry.start(
         deploymentId,
         DeploymentRegistry.ChangeBehavior.BLOCK,
         ZiplineServerDeploymentHandle::class.java,
         server
       )
+
+      // Keep the connection open by sending a message periodically
+      Timer("WebsocketHeartbeat", true).schedule(0, 10000) {
```

Comment:
This is terse, but the right way to do this since ~2004 is ScheduledExecutorService. Or perhaps something with a coroutine.